src/main/java/seedu/exercise/ui/CustomPropertyListPanel.java

```diff
@@ -18,15 +18,16 @@
 public class CustomPropertyListPanel extends UiPart<Region> {
     private static final String FXML = "CustomPropertyListPanel.fxml";
 
+    private final ObservableList<String> customPropertiesList;
+
     @FXML
     private ListView<String> customPropertiesListView;
 
     public CustomPropertyListPanel(Map<String, String> customPropertiesMap) {
         super(FXML);
-        ObservableList<String> customPropertiesList = toObservableList(customPropertiesMap);
+        this.customPropertiesList = toObservableList(customPropertiesMap);
```

Comment:
The `toObservableList` method should not be done by the UI. Some other component should convert it to a `ObservableList` like `model`. Ui should preferably not need to handle logic stuff

Reply:
Have created a method in CollectionsUtil that can help to convert a hashmap to a list of key-value strings. 

A method is introduced in Exercise to allow retrieval of an ``ObservableList<String>`` which contains key-value strings of the custom properties.